test/test_models.py

```diff
@@ -51,19 +51,26 @@ def _get_image(input_shape, real_image, device):
 
         img = Image.open(GRACE_HOPPER)
 
-        original_width, original_height = img.size
+        if weights is None:
```

Comment:
should we just pass the weights all the time? What's the reason for having them in only some cases but not all?

Reply:
In some cases the weight are really restrictive, for instance if we use `vit_h_14`, it will only accept the image_size of the size of the min_size of the weight: https://github.com/pytorch/vision/blob/main/torchvision/models/vision_transformer.py#L321 and in this case we can't do the test with lower resolution with the weight.

Also as of now, we dont use real weight for detection model test.

Reply:
> , it will only accept the image_size of the size of the min_size of the weight: https://github.com/pytorch/vision/blob/main/torchvision/models/vision_transformer.py#L321 and in this case we can't do the test with lower resolution with the weight

But isn't that a *good* thing? i.e. if we go below the `min_size` limit, wouldn't we expect the model to output garbage? And if not, why is the limit not lower?

Reply:
For test purpose, we might want to use smaller image even if the output is garbage but we can still check for the consistentcy (what we did so far with random image and random weight). And in this case if we set `weight=None` then it will basically behave like before, the get_image will assume that the test dont use real weight but rather initialized with random weight.